[jiangsuwangjing] iP

jiangsuwangjing opened the pull request:
# AlexPro

> “Your mind is for having ideas, not holding them.” – David Allen (source)

AlexPro frees your mind of having to remember things you need to do. It's:

- text-based
- easy to learn
- **FAST** _SUPER FAST_ to use

All you need to do is:

1. [Download it from here](https://github.com/jiangsuwangjing/ip).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉
5. And it is **FREE**!

## Features:
- Managing tasks
- Managing deadlines (coming soon)
- Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jiangsuwangjing committed:
Renamed Duke to Alex. Added greeting and exit messages.

jiangsuwangjing committed:
Implemented Echo Function

jiangsuwangjing committed:
Implemented storing and displaying text of user input

jiangsuwangjing committed:
Created Task and TaskList. Implemented marking and unmarking tasks.

jiangsuwangjing committed:
Implemented children of tasks. Organised command parsing. Used enum for switch.

jiangsuwangjing committed:
Tweaked interaction logic. Added testcases to check the creation of 3 children and marking function.

jiangsuwangjing committed:
Added delete function to TaskList. ArrayList has been used throughout.

jiangsuwangjing committed:
Created some exceptions and handled them in the main program., including invalid number format, out of bound, and wrong command.

jiangsuwangjing committed:
Use enumeration in handling commands. Input is parsed and Command is returned.

jiangsuwangjing committed:
Tweak response message when adding tasks and display list to meet expected behaviour.

jiangsuwangjing committed:
Fix bug of extra new line in welcome message.

jiangsuwangjing committed:
Fix extra line from welcome message.

jiangsuwangjing committed:
Add Ui class for more OOP. Change AlexExceptions for clearer error handling.

jiangsuwangjing committed:
Merge branch 'A-MoreOOP'

jiangsuwangjing committed:
Add Storage class to store data. Implement task method getSavedDataFormat() to show the format to be saved.

jiangsuwangjing committed:
Implement initialisation of data and deletion of task in Storage. Create DataException to deal with various exceptions that may occur during handling data file.

jiangsuwangjing committed:
Merge branch 'branch-Level-7'

jiangsuwangjing committed:
Add Command class and subclasses to deal with different command characteristics. Add Parser class to parse user input into Command, so that commands can be executed in a standardised procedure in main. Update Alex implementation by moving the running from main() to run().

jiangsuwangjing committed:
Merge branch 'branch-A-MoreOOP'

jiangsuwangjing committed:
Complete storage feature with saving marking and unmarking information.

jiangsuwangjing committed:
Merge branch 'branch-Level-7'

jiangsuwangjing committed:
no message

jiangsuwangjing committed:
Implement deadlines with LocalDate and print it in a different format.

jiangsuwangjing committed:
Handle the case when the date cannot be parsed.

jiangsuwangjing committed:
Merge branch 'branch-Level-8'

jiangsuwangjing committed:
Refactor the files and organise them into packages, consisting of command, exception and task.

jiangsuwangjing committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2425S2/ip

jiangsuwangjing committed:
Set up gradle and change code to pass gradle clean test.

jiangsuwangjing committed:
Add JUnit tests for TaskList and Parser.

jiangsuwangjing committed:
A-Jar

jiangsuwangjing committed:
Add JavaDoc for files in alex, alex.task, alex.exception. Tweak code to comply with coding standard.

jiangsuwangjing committed:
Implement find tasks by keyword function.

jiangsuwangjing committed:
Merge branch 'branch-Level-9'

jiangsuwangjing committed:
Merge branch 'branch-A-JavaDoc'

jiangsuwangjing committed:
Fix coding standard by changing switch indentation.

jiangsuwangjing committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/alex/Parser.java
Resolved comflict of new line.

jiangsuwangjing committed:
Configure CheckStyle with Gradle. Modify code to fit the coding standard.

jiangsuwangjing committed:
Merge branch 'A-CheckStyle'

jiangsuwangjing committed:
Incorporate GUI to Alex. Change UI into a interface.

jiangsuwangjing committed:
Merge branch 'branch-Level-10'

J4Joshua commented on src/main/java/alex/Ui.java:
```diff
@@ -0,0 +1,50 @@
+package alex;
```
Could be in the Ui package

J4Joshua commented on src/main/java/alex/Parser.java:
```diff
@@ -0,0 +1,110 @@
+            CommandType commandType;
+            commandType = Parser.extractCommand(inputStr);
+
+            switch (commandType) {
```
Good indentation for switch

J4Joshua commented on src/main/java/alex/Alex.java:
```diff
@@ -0,0 +1,55 @@
+        this.ui = ui;
+    }
+
+    /**
```
What is this for? Can remove i think

J4Joshua reviewed (commented):
LGTM just minor issues

jiangsuwangjing committed:
Add assert feature

Use assertions to document assumptions in various points in the code. The assertions added include checking for null values and range of index parameters and length of arrays carrying variables.

Parser is untouched as it is the first point of entry from user input and errors are mostly caused by users and are already handled.

jiangsuwangjing committed:
Refactor and improve code quality

Cleaned up the zombie code that are commented out or not used anywhere from Alex.

Extracted the logic of parsing command details from the Parser to the specific Command classes. This reduces the long parse function and adheres to SLAP. Even though extracting the commands with switch is still long, it is still necessary and may be improved in future iterations.

jiangsuwangjing committed:
Use stream for finding tasks

Used java Streams and filter to filter out the tasks containing the keywords.

jiangsuwangjing committed:
Merge pull request #1 from jiangsuwangjing/branch-A-Assertions

Add assert feature

jiangsuwangjing committed:
Merge commit 'b92dccaceb85f26405da3dd5ec145ab3f2c0b601' into branch-A-CodeQuality

jiangsuwangjing committed:
Merge pull request #2 from jiangsuwangjing/branch-A-CodeQuality

Refactor and improve code quality

jiangsuwangjing committed:
Merge pull request #3 from jiangsuwangjing/branch-A-Streams

Use stream for finding tasks

jiangsuwangjing committed:
Implement mass mark and unmark

Updated MarkCommand parser that identifies the range of indexes for the mass operation in the format of 'x-y' that represents the start and end of the tasks to mark.
Updated TaskList that handles mass mark and unmark operations.

jiangsuwangjing committed:
Implement mass delete

Updated parser function in DeleteCommand to parse the mass delete operation.

Created new method in Parser that parses the range from string representation 'x-y' into a pair of integer. This functionality is extracted because it is used in all the mass operations for better SLAP.

jiangsuwangjing committed:
Merge branch 'branch-C-MassOps'

jiangsuwangjing committed:
Update readme

Updated README and introduced basic functions.

jiangsuwangjing committed:
Minor bug fix and ui improvement

jiangsuwangjing committed:
Add javadoc

jiangsuwangjing committed:
Merge branch 'branch-A-CodeQuality'

jiangsuwangjing committed:
Add java dates in event tasks

jiangsuwangjing committed:
Update readme

jiangsuwangjing committed:
Add mock ui for testing
